kroxylicious-operator/src/main/java/io/kroxylicious/kubernetes/operator/ResourcesUtil.java

```diff
@@ -91,4 +96,25 @@ static Map<ResourceID, KafkaClusterRef> clusterRefs(Context<KafkaProxy> context)
                 .collect(Collectors.toMap(ResourceID::fromResource, Function.identity()));
     }
 
+    public static String name(@NonNull HasMetadata resource) {
+        return resource.getMetadata().getName();
+    }
+
+    public static String namespace(@NonNull HasMetadata resource) {
+        return resource.getMetadata().getNamespace();
+    }
+
+    public static Long generation(@NonNull HasMetadata resource) {
+        return resource.getMetadata().getGeneration();
+    }
+
+    public static String uid(@NonNull HasMetadata resource) {
+        return resource.getMetadata().getUid();
+    }
+
+    public static <T extends HasMetadata> Optional<T> resourceNamed(@NonNull Collection<T> collection, @NonNull String name) {
+        Objects.requireNonNull(collection);
+        Objects.requireNonNull(name);
+        return collection.stream().filter(item -> name(item).equals(name)).findFirst();
```

Comment:
Is `findFirst` the right behaviour here? Strictly speaking I think we need the k8s type of `T` here see [Object Names and IDs](https://kubernetes.io/docs/concepts/overview/working-with-objects/names/ ) in the k8s docs. 
> Each [object](https://kubernetes.io/docs/concepts/overview/working-with-objects/#kubernetes-objects) in your cluster has a [Name](https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#names) that is unique for that type of resource.

So as this is currently implemented a virtualcluster named `kafka` and a Strimzi Kafka cluster named `kafka` would both match this which could lead to some interesting inconsistencies...

I note the source of this refactor had an `orElseThrow` on the end probably for this reason.

Reply:
I might dodge that by renaming to `findOnlyResourceNamed` and have it throw on multiple matches

Reply:
the `orElseThrows()` is used in the ITs because we expect to get a match for the name. For this method I think it's better to offer up the Optional and let clients decide if it's a problem.